ci/dcos.sc

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env amm
+
+import ammonite.ops._
+import ammonite.ops.ImplicitWd._
+
+import scala.io.Source
+
+import $file.utils
+import utils.SemVer
+import utils.SemVerRead
+
+
+/**
+ * Main function for building DCOS registry file that works
+ * with DCOS 1.12+.
+ *
+ * @param version updated in the DCOS package files
+ */
+@main
+def buildRegistry(version: SemVer) = {
+  val universePath = createCatalogPackage(version)
+  val packagingTool = getPackagingTool()
+  buildDCOS(packagingTool, universePath, version)
+}
+
+/**
+ * Creates catalog package for Marathon with version info
+ * replaces __FILL_IN_VERSION_HERE__ with provided version
+ *
+ * @param version updated in the DCOS package files
+ * @param universePath location to build out the dcos universe files
+ */
+def createCatalogPackage(
+  version: SemVer,
+  universePath: Path = pwd / 'target / 'dcos
+): Path = {
+  rm! universePath
+  mkdir! universePath
+
+  // dcos catalog template directory
+  val templateDir: Path = pwd / 'dcos
+
+  // list all files in temp
+  templateDir.toIO.listFiles.foreach(file => {
+    val catalogFile = universePath / file.getName
+
+    Source.fromFile(file).getLines.map {
+      x => if(x.contains("__FILL_IN_VERSION_HERE__")) x.replace("__FILL_IN_VERSION_HERE__", version.toReleaseString) else x }
+      .foreach(x => write.append(catalogFile, x + System.lineSeparator()))
+    }
+  )
+  universePath
+}
+
+/**
+ * Pulls the registry build tool for the specific OS.
+ * Uses the marathon version which contains the "--use-local"
+ *
+ */
+def getPackagingTool(): Path = {
```

Comment:
I really would make this part of the `./ci/provision` as that file includes everything the a host requires to run a build. The same is true for `./ci/si_provision`. Leaving it here and calling it in `buildRegistry` makes it unsafe to use.

Reply:
I'm not ignore this... I think we may need this in other projects and I would like to reduce the number of files we need to link to... leaving this here encapsulates all of the registry tool work to this file.